server/lib/transactions.js

```diff
@@ -242,3 +247,126 @@ export async function sum(where) {
+  CONTRIBUTION: `Contribution`,
+  EXPENSE: `Expense`,
+  HOST_FEE: `Host Fee`,
+  HOST_FEE_SHARE: `Host Fee Share`,
+  HOST_FEE_SHARE_DEBT: `Host Fee Share Debt`,
+  PAYMENT_PROCESSOR_COVER: `Cover of Payment Processor Fee`,
+  PLATFORM_TIP: `Platform Tip`,
+  PLATFORM_TIP_DEBT: `Platform Tip Debt`,
+  PREPAID_PAYMENT_METHOD: `Prepaid Payment Method`,
+};
+
+export async function generateDescription(transaction, { req = null, full = false } = {}) {
+  let baseString = 'Transaction',
+    debtString = '',
+    tierString = '',
+    expenseString = '',
+    fromString = '',
+    toString = '';
+
+  if (transaction.isRefund && transaction.RefundTransactionId) {
+    const refundedTransaction = await (req
+      ? req.loaders.Transaction.byId.load(transaction.RefundTransactionId)
+      : models.Transactions.findByPk(order.RefundTransactionId));
+    if (refundedTransaction) {
+      const refundedTransactionDescription = await generateDescription(refundedTransaction, { req, full });
+      return `Refund of \`${refundedTransactionDescription}\``;
+    }
+  }
+
+  let order, expense, subscription, tier;
+
+  if (transaction.OrderId) {
+    order = await (req
+      ? req.loaders.Order.byId.load(transaction.OrderId)
+      : models.Orders.findByPk(transaction.OrderId));
+  }
+
+  if (kindStrings[transaction.kind]) {
+    baseString = kindStrings[transaction.kind];
+  }
+
+  if (transaction.kind === CONTRIBUTION) {
+    if (order && order.TierId) {
+      tier = await (req ? req.loaders.Tier.byId.load(order.TierId) : models.Tiers.findByPk(order.TierId));
+    }
+    if (order && order.SubscriptionId) {
+      subscription = await (req
+        ? req.loaders.Subscription.byId.load(order.SubscriptionId)
+        : models.Subscriptions.findByPk(order.SubscriptionId));
+    }
+    if (subscription && subscription.interval === 'month') {
+      baseString = `Monthly contribution`;
+    } else if (subscription && subscription.interval === 'year') {
+      baseString = `Yearly contribution`;
+    } else if (tier && tier.type === TICKET) {
+      baseString = `Registration`;
+    }
+    if (tier) {
+      tierString = ` (${tier.name})`;
+    }
+  } else if (transaction.kind === EXPENSE) {
+    if (transaction.ExpenseId) {
+      expense = await (req
+        ? req.loaders.Expense.byId.load(transaction.ExpenseId)
+        : models.Expenses.findByPk(transaction.ExpenseId));
+    }
+    if (expense) {
+      expenseString = ` - ${expense.description}`;
+    }
+  }
+
+  const account = await (req
+    ? req.loaders.Collective.byId.load(transaction.CollectiveId)
+    : models.Collectives.findByPk(order.CollectiveId));
+  const oppositeAccount = await (req
+    ? req.loaders.Collective.byId.load(transaction.FromCollectiveId)
+    : models.Collectives.findByPk(order.FromCollectiveId));
+
+  if (transaction.isDebt) {
+    debtString = ' owed';
+    if (transaction.type === CREDIT) {
+      if (full) {
+        toString = ` by ${account.name.trim()}`;
```

Comment:
`name` is supposed to be [trimmed](https://github.com/opencollective/opencollective-api/blob/776010bc0b0b39302778933d6269c22a445288a0/server/models/Collective.js#L161) on the model, is it not always the case? Maybe we can rather migrate if there are past entries that we want to fix.

Reply:
I saw some occurrences of space at the end.